File: 프로그래머스/lv2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    let answer = [];
    let daying = 0;
    let d = 0;
    
    for (let i = 0; i < progresses.length; i++) {
        let leftP = 100 - progresses[i]
        let leftDay = Math.ceil(leftP / speeds[i])
        
        if(leftDay > daying) {
            if(i !== 0) d += 1;
            daying = leftDay;
        }   
        answer[d] = answer[d] ? ++answer[d] : 1
    }
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math

def solution(progresses, speeds):
    answer = []
    queue = []
    for i in range(len(progresses)):
        queue.append(math.ceil((100 - progresses[i]) / speeds[i]))
    
    cnt = 0
    prev = queue[0]
    for i in range(1, len(queue)):
        if prev < queue[i]:
            answer.append(cnt + 1)
            prev = queue[i]
            cnt = 0
        else:
            cnt += 1
    answer.append(cnt + 1)
    
    return answer